util-images/network/netperfbenchmark/pkg/worker/worker.go

```diff
@@ -0,0 +1,458 @@
+type metricResponse struct {
+	Result          []float64
+	WorkerStartTime string
+	Error           string
+}
+
+//http  listen ports
+const (
+	workerListenPort = "5003"
+	httpPort         = "5301"
+)
+
+//Protocols supported
+const (
+	protocolTCP  = "TCP"
+	protocolUDP  = "UDP"
+	protocolHTTP = "HTTP"
+)
+
+//Start worker
+func Start() {
+	listenToCntrlr()
+}
+
+func listenToCntrlr() {
+	h := map[string]func(http.ResponseWriter, *http.Request){"/startTCPServer": StartTCPServer,
+		"/startTCPClient": StartTCPClient, "/startUDPServer": StartUDPServer, "/startUDPClient": StartUDPClient,
+		"/startHTTPServer": StartHTTPServer, "/startHTTPClient": StartHTTPClient, "/metrics": Metrics}
+	startListening(workerListenPort, h)
+	klog.Info("Started listening to Server")
+}
+
+func startListening(port string, handlers map[string]func(http.ResponseWriter, *http.Request)) {
+	klog.Info("In StartHTTPServer")
+	for p, h := range handlers {
+		http.HandleFunc(p, h)
+	}
+	err := http.ListenAndServe(":"+port, nil)
+	if err != nil {
+		klog.Fatalf("Failed starting http server for port: %v, Error: %v", port, err)
+	}
+}
+
+//Metrics returns the metrics collected
+func Metrics(res http.ResponseWriter, req *http.Request) {
+	var reply metricResponse
+	klog.Info("In metrics")
+	var stat string
+	select {
+	case stat = <-resultStatus:
+		if stat != "OK" {
+			klog.Error("Error collecting metrics:", stat)
+			reply.Error = "metrics collection failed:" + stat
+			createResp(reply, &res)
+			return
+		}
+		klog.Info("Metrics collected")
+	default:
+		klog.Info("Metric collection in progress")
+		reply.Error = "metric collection in progress"
+		createResp(reply, &res)
+		return
+	}
+	o, err := parseResult()
+	if err != nil {
+		reply.Error = err.Error()
+		createResp(reply, &res)
+		return
+	}
+	reply.Result = o
+	//startime and other variables below can be removed in future, for debugging during initial runs
+	stStr := strconv.FormatInt(startedAt, 10)
+	ftStr := strconv.FormatInt(futureTime, 10)
+	dtStr := strconv.FormatInt(futureTime-startedAt, 10)
+	reply.WorkerStartTime = "StartedAt:" + stStr + " FutureTime:" + ftStr + " diffTime:" + dtStr
+	createResp(reply, &res)
+}
+
+func createResp(resp interface{}, w *http.ResponseWriter) {
+	klog.Info("Inside reply")
+	(*w).Header().Set("Content-Type", "application/json")
+	(*w).WriteHeader(http.StatusOK)
+	klog.Info("Marshalled Resp:", resp)
+	b, err := json.Marshal(resp)
+	if err != nil {
+		klog.Info("Error marshalling to json:", err)
+	}
+	(*w).Write(b)
+}
+
+//StartTCPClient starts iperf client for tcp measurements
+func StartTCPClient(res http.ResponseWriter, req *http.Request) {
+	klog.Info("In StartTCPClient")
+	klog.Info("Req:", req)
+	ts, dur, destIP, _ := parseURLParam(req)
+	if dur == "" || destIP == "" {
+		createResp(startWrkResponse{error: "missing/invalid required parameters"}, &res)
+		return
+	}
+	go schedule(protocolTCP, ts, dur,
+		"iperf", []string{"-c", destIP, "-f", "K", "-l",
```

Comment:
please add a comment explaining the parameters, something like this would work:
```go
// -c destIP connect to destIP
// -f K format Kbits
// -l 20 read/write buffer size 20 butes
// -b 1M bandwidth 1 Mbits
// -i 1 repert stats every 1 sec
// -t dur run durring dur seconds
```

Reply:
The modified code has commands for all protocols declared as part of variable declaration together. Hence have put comment for each tool type with all options we have used. Please let me know if that would suffice.